### PR TITLE
Review Dockerfile and RPM files for building odh-kf-notebook-controller and odh-notebook-controller in Konflux

### PR DESCRIPTION
**Pull request by MohammadiIram**
The issue resolved by this Pull Request:
Resolves #https://issues.redhat.com/browse/RHOAIENG-14097

### CODE REVIEW

**Comment by atheo89** [comment]
Hi @MohammadiIram could you rebase this PR to `main` branch, main is our default?

**Comment by jiridanek** [comment]
Please explain why this commit cannot go to opendatahub-io, as per objection at https://github.com/red-hat-data-services/odh-dashboard/pull/356#issuecomment-2399741127

**Comment by atheo89** [comment]
>Please explain why this commit cannot go to opendatahub-io, as per objection at https://github.com/red-hat-data-services/odh-dashboard/pull/356#issuecomment-2399741127

Is Konflux intended solely for downstream repositories? It would be helpful to have more information and context about this PR. @MohammadiIram

cc: @harshad16

File: components/notebook-controller/Dockerfile.konflux [hunk 1]
```diff
@@ -46,21 +45,27 @@ RUN microdnf install -y shadow-utils &&\
...
+
+
+## Switch to a non-root user
+USER 1001:0
+
+ENTRYPOINT [ "/manager" ]
 
-LABEL com.redhat.component="odh-kf-notebook-controller-container" \
-      name="managed-open-data-hub/odh-kf-notebook-controller-rhel8" \
+LABEL com.redhat.component="odh-notebook-controller-container" \
```
Comment (atheo89) [inline components/notebook-controller/Dockerfile.konflux:61]: Could you keep the naming of the notebook controllers like:

- notebook-controller -> kf-notebook-controller (Is the notebook controller that comes from kubeflow)
- odh-notebook-controller-> odh-notebook-controller (Is the odh mutated notebook controller)

This is to keep the naming consistency among the project.

**Comment by MohammadiIram** [comment]
> > Please explain why this commit cannot go to opendatahub-io, as per objection at [red-hat-data-services/odh-dashboard#356 (comment)](https://github.com/red-hat-data-services/odh-dashboard/pull/356#issuecomment-2399741127)
> 
> Is Konflux intended solely for downstream repositories? It would be helpful to have more information and context about this PR. @MohammadiIram
> 
> cc: @harshad16


@atheo89 Yes, for now, it is intended to remain only in the RHDS repository. We cannot merge it into ODH due to the UBI digests being consistently updated.

File: components/notebook-controller/Dockerfile.konflux (hunk 1 above)
Comment (MohammadiIram) [reply components/notebook-controller/Dockerfile.konflux:61]: This is the same label we used in midstream when building with CPaaS. I’d like to know if changing the label would have any other effects?

**Comment by dchourasia** [comment]
@atheo89 @harshad16 I have sent an email with details of why we want to keep these files to downstream for 2.16 version, please let me know if you have any queries.

File: components/notebook-controller/Dockerfile.konflux (hunk 1 above)
Comment (jiridanek) [reply components/notebook-controller/Dockerfile.konflux:61]: I think we should not change the label, surely not in this PR.

**Review by jiridanek — APPROVED** [review]
I'd like to see a konflux build that builds this, see the build logs, and try out the resulting image. But just looking at the files

/lgtm

File: components/notebook-controller/Dockerfile.konflux (hunk 1 above)
Comment (atheo89) [reply components/notebook-controller/Dockerfile.konflux:61]: Thank you @MohammadiIram for explain this. That's fine for me then

**Pull request merged**